src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1290,17 +1290,24 @@ public StringBuilder Insert(int index, string? value)
             return this;
         }
 
-        public StringBuilder Insert(int index, bool value) => Insert(index, value.ToString(), 1);
+#pragma warning disable CA1830 // Prefer strongly-typed Append and Insert method overloads on StringBuilder. No need to fix for the builder itself
+        public StringBuilder Insert(int index, bool value) => Insert(index, value.ToString());
+#pragma warning restore CA1830
 
         [CLSCompliant(false)]
-        public StringBuilder Insert(int index, sbyte value) => Insert(index, value.ToString(), 1);
+        public StringBuilder Insert(int index, sbyte value) => InsertSpanFormattable(index, value);
 
-        public StringBuilder Insert(int index, byte value) => Insert(index, value.ToString(), 1);
+        public StringBuilder Insert(int index, byte value) => InsertSpanFormattable(index, value);
 
-        public StringBuilder Insert(int index, short value) => Insert(index, value.ToString(), 1);
+        public StringBuilder Insert(int index, short value) => InsertSpanFormattable(index, value);
 
         public StringBuilder Insert(int index, char value)
         {
+            if ((uint)index > (uint)Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index), SR.ArgumentOutOfRange_Index);
```

Comment:
Is the throw worth moving into a throw helper?

Reply:
Better for another time; throw helpers are very rarely used in `StringBuilder`.